Library/Homebrew/diagnostic.rb

```diff
@@ -122,7 +122,8 @@ def examine_git_origin(repository_path, desired_origin)
             properly. You can solve this by adding the remote:
               git -C "#{repository_path}" remote add origin #{Formatter.url("https://github.com/#{desired_origin}.git")}
           EOS
-        elsif !current_origin.match?(%r{#{desired_origin}(\.git|/)?$}i)
+        elsif !current_origin.match?(%r{#{desired_origin}(\.git|/)?$}i) && \
+              [HOMEBREW_BREW_REMOTE, HOMEBREW_CORE_REMOTE].none?(current_origin)
```

Comment:
It would be preferable to pass the relevant `HOMEBREW_BREW_REMOTE` or `HOMEBREW_CORE_REMOTE` as something like `user_origin` or just use pass `HOMEBREW_BREW_REMOTE` or `HOMEBREW_CORE_REMOTE` as the `desired_origin` value. Otherwise you'll allow the Homebrew/brew remote to be set to the Homebrew/homebrew-core remote (or vice versa) which has happened in the past and is one of the reasons this check is here.